addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/SOAPMsgConfig.java

```diff
@@ -104,8 +104,17 @@ public void setBindOp(BindingOperation bindOp) {
         this.bindOp = bindOp;
     }
 
-    public boolean equals(SOAPMsgConfig other) {
-        return this.getHash() == other.getHash();
+    @Override
+    public boolean equals(Object other) {
+        if (other == null || !(other instanceof SOAPMsgConfig)) {
+            return false;
+        }
+        return this.getHash() == ((SOAPMsgConfig) other).getHash();
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
Worth checking that this will not affect existing usages.

Reply:
The method is not used within the extension, so as far as I can see it should be fine.

Reply:
Even indirectly? (e.g. when the config is added to a `HashMap`, or other collections that make use of the hash.)

Reply:
Yup that's definitely possible. There are hash based collections in use, however they also seem to be unit tested.